src/table/boot.rs

```diff
@@ -443,6 +447,9 @@ pub enum MemoryType {
     PalCode,
     /// Memory region which is usable and is also non-volatile.
     PersistentMemory,
+    // SAFETY: UEFI defines a MaxMemoryType, therefore adding new memory types
```

Comment:
Unfortunately, quoting from the spec:
> MemoryType values in the range
0x70000000..0x7FFFFFFF are reserved for OEM use.

> MemoryType values in the range 0x80000000..0xFFFFFFFF are reserved for use by
UEFI OS loaders that are provided by operating system vendors.

Reply:
Ah, so UEFI max values can lie... Sad, sad... Let's review the whole thing in this new light then!

Reply:
Fixed in the latest commit. I checked the GOP case again, and the spec is more insistent on the fact that "Valid EFI_GRAPHICS_PIXEL_FORMAT enum values are less than this value", so I still feel confident leaving it alone. Of course, there would also be a consistency argument in favor of making every enum used in FFI a newtype_enum now that these are becoming quite common...